serializable_request.py

```diff
@@ -1,5 +1,48 @@
 import requests
 import json
+from urllib.parse import urlsplit, urlunsplit
+
+
+class Response():
+    def __init__(self, status, data, headers):
+        self.status = status
+        self.data = data
+        self.headers = headers
+
+    @classmethod
+    def from_file(cls, fname):
+        with open(fname, 'r') as f:
+            return cls.from_json(f.read())
+
+    @classmethod
+    def from_json(cls, json_data):
+        data = json.loads(json_data)
+        return cls(
+            int(data['status']),
+            data['data'].encode('utf-8'),
+            data['headers'],
+        )
+
+    @classmethod
+    def from_requests_response(cls, res: requests.Response):
+        return cls(res.status_code, res.content, dict(res.headers))
+
+    def to_file(self, fname):
+        with open(fname, 'w') as f:
+            f.write(self.as_json())
+
+    def as_json(self):
```

Comment:
```suggestion
    def to_json(self):
```
Why is it `to_file` but `as_json`? Let's be consistent.

Reply:
You'll be surprised, but this is not an accident : )

I'm a big fan of command-query separation. `as` is a query, value is returned, `to` is a command, nothing is returned, some action is performed.
If you think this is not a good reason, I'll change this.

Reply:
That's a really interesting point you made. I've got to think about this. Honestly, I haven't meet with this approach before. I think most Python libraries follow the `to_` convention. I've also seen `to_` vs `as_` used in other languages to distinguish between creating a new object and returning a reference/view of the existing object but I guess it doesn't apply to Python.

Reply:
There is e.g. `dataclassess.asdict()/astuple()`.
There could be also `Request.as_file()` as a function that creates any file and returns filename.

(not insisting at all)